### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by siarez** [1adrianb/face-alignment#128]
First, thank you for all the work and sharing the code.
Currently, I'm using additional data to fine-tune the network, and I'm facing a strange situation where **the training loss goes down, but the training pixel-MSE is goes up** slightly. [Here is an image of the curves](https://pasteboard.co/I55Ks6L.png). 

- The model is being trained on a P100 Pascal GPU. 
- I'm using PyTorch 1.0.
- I have experimented with freezing the first 3 hourglass modules with no luck. 

I have been debugging the code for 4 days! And I'm still stuck. Any pointers is appreciated.

**Comment by siarez** [comment]
For reasons that I'm not entirely sure of yet, using a older version of the `get_preds_fromhm` from the commit history doesn't have this issue. Basically, the commit that address issue #32 is causing this issue.
I'm guessing the changes to `get_preds_fromhm` should have been accompanied by changes to `draw_gaussian`, but I'm not sure exactly how. Because these two functions need to be inverses of each other, but the the commit that address issue #32, broke that symmetry.

**Comment by siarez** [comment]
**Recap:**
Something strange happens during fine-tuning. That is as the loss decreases, pixel MSE increases.
The older version of the `get_preds_fromhm` function, before this [commit](https://github.com/1adrianb/face-alignment/commit/a36efd3cbf80d79af60248bbb68a3f77aa99ef32) doesn't have this problem.
While the new version of the `get_preds_fromhm` gets significantly better accuracy, fine-tuning using it results in pixel MSE getting worse as loss decreases.

**Update:**
During fine-tuning `create_target_heatmap` is used to create target heatmaps from ground truth landmarks. 

I think the reason for pixel MSE increasing is that the `get_preds_fromhm` and `create_target_heatmap` are not inverses of each other, after that commit. I think the [updates](https://github.com/1adrianb/face-alignment/commit/a36efd3cbf80d79af60248bbb68a3f77aa99ef32) to `get_preds_fromhm` should have been accompanied by updates to `create_target_heatmap`. But
`create_target_heatmap` has remain unchanged after the update to `get_preds_fromhm`

I think the reason fine-tuning the older version (before the commit) doesn't have that problem, is that in the old version, these two functions are inverses of each other. But that was broken when `get_preds_fromhm` was updated.

I can't quite wrap my head around all the transforms that occurs inside these functions, so I'm not sure how to fix it. Any help is appreciated.

**Comment by siarez** [comment]
Turns out the issue was how I was creating target heatmaps from ground truth landmarks. The transforms were confusing me. I'll submit a pool request with my function for heatmap creation which may be of interest for anyone who want to train the model in Pytorch.

**Comment by wqz960** [comment]
Hi @siarez  have you trained your code on 300W. I want to train a model on 300W-2D dataset, have you try that?

**Comment by siarez** [comment]
I actually used my code to train (rather fine-tune) on our own internal dataset. Our internal dataset is also 2D. I'll try to create a public repository for training once my code is cleaned up. For now, you'll need to write your own `fit` and `validation` functions that use the `create_target_heatmap` function in my pull request to create target heat-maps from your ground truth landmarks.
